android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/UserLocationView.java

```diff
@@ -574,7 +574,7 @@ private void setLocation(Location location) {
             mShowDirection = true;
             mGpsMarkerDirection = 0;
             if (location.hasBearing()) {
-                mMapView.setBearing(mUserLocation.getBearing());
+                mMapView.setBearing(location.getBearing());
```

Comment:
@zugaldia I'm wondering if it might make more sense to move `mUserLocation = location` up in the method instead.  [Currently](https://github.com/mapbox/mapbox-gl-native/blob/a0f2553ab4af7e9723e1e0723bb070f665321aeb/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/UserLocationView.java#L591) it's after this code.